pkg/aggregator/aggregator.go

```diff
@@ -732,7 +737,23 @@ func (agg *BufferedAggregator) run() {
 			agg.hostname = h
 			changeAllSendersDefaultHostname(h)
 			agg.hostnameUpdateDone <- struct{}{}
+		case orchestratorMetadata := <-agg.orchestratorMetadataIn:
+			aggregatorOrchestratorMetadata.Add(1)
+			// each resource has its own payload so we cannot aggregate
+			// use a routine to avoid blocking the aggregator
+			go func(orchestratorMetadata senderOrchestratorMetadata) {
+				err := agg.serializer.SendOrchestratorMetadata(
+					orchestratorMetadata.msgs,
+					agg.hostname,
+					orchestratorMetadata.clusterID,
+					orchestratorMetadata.payloadType,
+				)
+				if err != nil {
+					log.Errorf("Error submitting orchestrator data: %s", err)
+				}
+			}(orchestratorMetadata)
```

Comment:
the metadata are already "full" messages for a specific kubernetes type. So we expect calling this function X times every `collection_interval` seconds, where X is the number of resource type we collect (so currently 5 times every 10 seconds). As they are already "aggregated" messages by resource type we can't really re-aggregate them especially because only this check is sending them and already doing the work:

Let's take `deployments` as an example, they are processed here https://github.com/DataDog/datadog-agent/blob/1e00079b02c20392de505e2c6b711170cabf2956/pkg/collector/corechecks/cluster/orchestrator_processor.go#L64-L75 and aggregated in group of message bodies.

then the list of message (containing list of deployments) are sent here https://github.com/DataDog/datadog-agent/blob/1e00079b02c20392de505e2c6b711170cabf2956/pkg/collector/corechecks/cluster/orchestrator.go#L271

Does it sound reasonable?

Reply:
Does this scale with the cluster ? Will it be 5 time no matter the number of pod/container/deployments... ?

I just want to make sure we're not blocking the aggregator for too long, preventing it from receiving dogstatsd metrics. If it's 5 times every 10s, I think we're good.

Reply:
it should be called 5 times no matter the size of the cluster as all resources are already packed within the list of messages, this number will grow slowly as we collect more resources but will  stay around 20/30 max (talking long term here, mid term probably around 10-15)

> I just want to make sure we're not blocking the aggregator for too long

Isn't it the goal of the routine to avoid blocking the aggregator?

>  preventing it from receiving dogstatsd metrics

note that the cluster-agent or dedicated cluster check runners (the 2 cases where the orchestrator forwarder is stared and this part of the code would be called) should not be able to receive any dsd traffic

Reply:
> it should be called 5 times no matter the size of the cluster 

Great, then we can definitely afford the goroutine each time.

> Isn't it the goal of the routine to avoid blocking the aggregator?

The more goroutine we run the bigger performance impact we have

> note that the cluster-agent or dedicated cluster check runners (the 2 cases where the orchestrator forwarder is stared and this part of the code would be called) should not receive any dsd traffic

￼good to know !